server/src/main/java/io/crate/execution/ddl/tables/AddColumnTask.java

```diff
@@ -82,11 +82,12 @@ protected ClusterState execute(ClusterState currentState, AddColumnRequest reque
         DocTableInfo currentTable = docTableInfoFactory.create(request.relationName(), currentState);
 
         boolean allExist = true;
+        List<Reference> newColumns = new ArrayList<>();
```

Comment:
I would unite it with `addMissingParentColumns` to avoid copying request twice - just did it that way to split into 2 commits and highlight 2 different issues. Basically would have smth like `addMissingParentColumnsAndRemoveExistingOnes` - need to come up with a sorter name.

Reply:
Sounds good.

Maybe `normalizeColumns` ? It's abstract but I think with some method javadocs that mentions it adds missing parents and removes existing ones it's fine.

Reply:
BTW, `test_skip_existing_columns_but_handle_all_others` fails if I removed second commit, i.e we don't have break but don't remove existing columns. This way we just try to insert all existing/non-exisintg columns.

I expected it to be maybe not optimal but rewrite existing columns. However, version without second commit fails with
```
java.lang.IllegalArgumentException: Mapper for [x] conflicts with existing mapping:
[mapper [x] has different [doc_values] values]
```

Should we be bothered with this non-overwriting-exisitng behavior? Does it highlight any other issue?

Reply:
OK, it's not an issue on master, doc-values comes as true in integration test - it was coming to AddColumnTask as false since we create SimpleReference with explicit docValue false ctor

Reply:
It should never override an existing column, but mapping conflict detection would be good to have. In that case a insert should probably retry to get the new schema.

Reply:
> It should never override an existing column

it will not as we remove existing ones. Pushed an update, could you have a look?

Reply:
>  should never override an existing column, but mapping conflict detection would be good to have

Yes, I think keeping existing columns is fine - we don't override since we merge maps and we will do conflict detection this way. 

I had an assumption that merging same column twice can cause flakiness but since we merge maps it will not override anything.  I kept only don't-break-early